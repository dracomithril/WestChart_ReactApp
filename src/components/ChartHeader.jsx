/**
 * Created by XKTR67 on 5/25/2017.
 */
import React from "react";
import PropTypes from "prop-types";
import {Button, ButtonGroup, Label, OverlayTrigger, Popover} from "react-bootstrap";
import SongsPerDay from "./SongsPerDay";
import FilteringOptions from "./FilteringOptions";
import PickYourDate from "./PickYourDate";
import "./components.css";
let utils = require('./../utils');

export default class ChartHeader extends React.Component {
    render() {
        const {store} = this.context;
        const {since, until, last_update} = store.getState();
        const {count, error_days} = this.props;
        const options = {weekday: 'short', month: '2-digit', day: 'numeric'};
        return (<div className="chartHeader">
            <div className="dock1">
                <SongsPerDay error_days={error_days}/>
                <PickYourDate/>
            </div>
            <small>{count}</small>
            <div className="chartButtons">
                <ButtonGroup bsSize="large">
                    <OverlayTrigger trigger={["hover", "focus"]} placement="top" overlay={<Popover id="update_info">
                        <span>{"since: "}</span>
                        <Label
                            bsStyle="success">{since !== '' ? new Date(since).toLocaleDateString('pl-PL', options) : 'null'}</Label>
                        <span>{" to "}</span>
                        <Label
                            bsStyle="danger">{until !== '' ? new Date(until).toLocaleDateString('pl-PL', options) : 'null'}</Label><br/>
                        <small
                            id="updateDate">{' Last update: ' + new Date(last_update).toLocaleString('pl-PL')}</small>
                    </Popover>}>
                        <Button id="updateChartB" onClick={() => utils.UpdateChart(store)}
                                bsStyle="primary">Update</Button>
                    </OverlayTrigger>
                    <FilteringOptions/>
                </ButtonGroup>
            </div>

        </div>)
    }
}
ChartHeader.contextTypes = {
    store: PropTypes.object
};
