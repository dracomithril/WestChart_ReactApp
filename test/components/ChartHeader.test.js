/**
 * Created by Gryzli on 18.06.2017.
 */
import React from 'react';
import PropTypes from 'prop-types';
import ChartHeader from './../../src/components/ChartHeader';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
const initial_state =require('./../data/initial_state.json');
const mockStore = configureMockStore([]);
describe('<ChartHeader/>', () => {
    it('renders without crashing ChartPresenter', () => {
        const store = mockStore(initial_state);
        const wrapper = shallow(
            <ChartHeader error_days={[]}/>, {
                context: {store},
                childContextTypes: {store: PropTypes.object}
            }
        );
        expect(shallowToJson(wrapper)).toMatchSnapshot();

    });

});