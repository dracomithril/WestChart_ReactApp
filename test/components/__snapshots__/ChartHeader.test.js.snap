// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChartHeader/> renders without crashing ChartPresenter 1`] = `
<div
  className="chartHeader"
>
  <div
    className="dock1"
  >
    <SongsPerDay
      error_days={Array []}
    />
    <PickYourDate />
  </div>
  <div
    className="chartButtons"
  >
    <OverlayTrigger
      defaultOverlayShown={false}
      overlay={
        <Popover
          bsClass="popover"
          id="update_info"
          placement="right"
        >
          <span>
            since: 
          </span>
          <Label
            bsClass="label"
            bsStyle="success"
          >
            Invalid Date
          </Label>
          <span>
             to 
          </span>
          <Label
            bsClass="label"
            bsStyle="danger"
          >
            Sat, 03/11
          </Label>
          <br />
          <small
            id="updateDate"
          >
             Last update: Invalid Date
          </small>
        </Popover>
      }
      placement="top"
      trigger={
        Array [
          "hover",
          "focus",
        ]
      }
    >
      <div>
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="primary"
          disabled={false}
          id="updateChartB"
          onClick={[Function]}
        >
          Update
        </Button>
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="success"
          disabled={false}
          id="quickSummary"
          onClick={[Function]}
        >
          Quick summary
        </Button>
      </div>
    </OverlayTrigger>
  </div>
  <FilteringOptions />
</div>
`;
